server/src/main/java/org/elasticsearch/cluster/routing/allocation/allocator/DesiredBalanceComputer.java

```diff
@@ -54,15 +54,27 @@ public DesiredBalance compute(
 
         final var routingAllocation = desiredBalanceInput.routingAllocation().mutableCloneForSimulation();
         final var routingNodes = routingAllocation.routingNodes();
-        final var ignoredShards = new HashSet<>(desiredBalanceInput.ignoredShards());
         final var changes = routingAllocation.changes();
+        final var ignoredShards = desiredBalanceInput.ignoredShards();
         final var knownNodeIds = routingAllocation.nodes().stream().map(DiscoveryNode::getId).collect(Collectors.toSet());
         final var unassignedPrimaries = new HashSet<ShardId>();
 
         if (routingNodes.isEmpty()) {
             return new DesiredBalance(desiredBalanceInput.index(), Map.of());
         }
 
+        // start shards that will be initialized
+        for (final var iterator = routingNodes.unassigned().iterator(); iterator.hasNext();) {
+            final var shardRouting = iterator.next();
+            var info = shardRouting.unassignedInfo();
+            if (info != null
+                && info.getLastAllocatedNodeId() != null
+                && routingNodes.node(info.getLastAllocatedNodeId()) != null
+                && ignoredShards.contains(shardRouting) == false) {
```

Comment:
Should we do it even if shard is ignored?

Reply:
I am writing a unit test for this

Reply:
I think we _only_ want to do this if a shard is ignored because that indicates the `GatewayAllocator` is still taking responsibility for it (and therefore we expect will eventually assign it). If the shard is not ignored then its last-allocated ID shouldn't be relevant, it's effectively a new shard and should be left for the delegate allocator.